// devextreme-angular/esm2022/ui/popup/service/service.component.mjs
/*!
 * devextreme-angular
 * Version: 24.2.3
 * Build date: Fri Dec 06 2024
 *
 * Copyright (c) 2012 - 2024 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
import { Component, ElementRef, Inject, NgZone, PLATFORM_ID, TransferState, Type, ViewChild, } from '@angular/core';
import { DxTemplateHost, IterableDifferHelper, NestedOptionHost, WatcherHelper, } from 'devextreme-angular/core';
import { DxPopupComponent } from '../component';
import { DxServicePopupInsertionDirective } from './insertion.directive';
import * as i0 from "@angular/core";
import * as i1 from "devextreme-angular/core";
export class PopupServiceComponent extends DxPopupComponent {
    contentComponent;
    popupOptions;
    contentInsertion;
    contentRef;
    constructor(contentComponent, popupOptions, elementRef, ngZone, templateHost, _watcherHelper, _idh, optionHost, transferState, platformId) {
        super(elementRef, ngZone, templateHost, _watcherHelper, _idh, optionHost, transferState, platformId);
        this.contentComponent = contentComponent;
        this.popupOptions = popupOptions;
    }
    ngAfterViewInit() {
        super.ngAfterViewInit();
        if (this.popupOptions) {
            this.instance.option(this.popupOptions);
        }
        this.contentRef = this.contentInsertion?.viewContainerRef.createComponent(this.contentComponent);
    }
    /** @nocollapse */ static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PopupServiceComponent, deps: [{ token: 'popupServiceContentComponent' }, { token: 'popupServiceOptions' }, { token: i0.ElementRef }, { token: i0.NgZone }, { token: i1.DxTemplateHost }, { token: i1.WatcherHelper }, { token: i1.IterableDifferHelper }, { token: i1.NestedOptionHost }, { token: i0.TransferState }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Component });
    /** @nocollapse */ static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: PopupServiceComponent, isStandalone: true, selector: "ng-component", providers: [
            DxTemplateHost,
            WatcherHelper,
            NestedOptionHost,
            IterableDifferHelper,
        ], viewQueries: [{ propertyName: "contentInsertion", first: true, predicate: DxServicePopupInsertionDirective, descendants: true }], usesInheritance: true, ngImport: i0, template: '<ng-template popup-content-insertion></ng-template>', isInline: true, dependencies: [{ kind: "directive", type: DxServicePopupInsertionDirective, selector: "[popup-content-insertion]" }] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PopupServiceComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    imports: [DxServicePopupInsertionDirective],
                    providers: [
                        DxTemplateHost,
                        WatcherHelper,
                        NestedOptionHost,
                        IterableDifferHelper,
                    ],
                    template: '<ng-template popup-content-insertion></ng-template>',
                }]
        }], ctorParameters: () => [{ type: i0.Type, decorators: [{
                    type: Inject,
                    args: ['popupServiceContentComponent']
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['popupServiceOptions']
                }] }, { type: i0.ElementRef }, { type: i0.NgZone }, { type: i1.DxTemplateHost }, { type: i1.WatcherHelper }, { type: i1.IterableDifferHelper }, { type: i1.NestedOptionHost }, { type: i0.TransferState }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }], propDecorators: { contentInsertion: [{
                type: ViewChild,
                args: [DxServicePopupInsertionDirective]
            }] } });
